File: src/main/java/voidchess/player/ki/SearchTreePruner.java
/*
 * Created on 05.11.2006
 */


package voidchess.player.ki;

public interface SearchTreePruner {

    boolean stopMinDynamicEvaluation(int depth, boolean thisMove_isChess, boolean thisMove_hasHitFigure,
                                     boolean lastMove_isChess, boolean lastMove_hasHitFigure);

    boolean stopMaxDynamicEvaluation(int depth, boolean thisMove_isChess, boolean thisMove_hasHitFigure,
                                     boolean lastMove_isChess, boolean lastMove_hasHitFigure);

}

File: src/main/kotlin/voidchess/player/ki/SearchTreePruner.kt
package voidchess.player.ki

interface SearchTreePruner {

    fun stopMinDynamicEvaluation(depth: Int, thisMove_isChess: Boolean, thisMove_hasHitFigure: Boolean,
                                 lastMove_isChess: Boolean, lastMove_hasHitFigure: Boolean): Boolean

    fun stopMaxDynamicEvaluation(depth: Int, thisMove_isChess: Boolean, thisMove_hasHitFigure: Boolean,
                                 lastMove_isChess: Boolean, lastMove_hasHitFigure: Boolean): Boolean

}